### PR TITLE
Arrow Reflect Pannel

### LINKED ISSUE #58: panel that reflects arrows

### PR DESCRIPTION
**Pull request by chmilk2**
Adds a script that can be attached to any object with a collider that reflects the velocity of any object that collides with it. Also has Serialized Fields that control a multiplier of exit velocity and a boost to the Y velocity
closes #58

### CODE REVIEW

File: Assets/Scripts/ArrowReflector.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    private void OnCollisionEnter(Collision collision)
+    {
+        //defines varibles to edit collision object
+        var rigidBody = collision.rigidbody;
+        var velocity = rigidBody.velocity;
+
+        //adds the velocity and normalized venocity to get a good approxamation of the reflection angle,
+        //then multiplies by the velocites magnatude and energy multipyer to get resulting force
+        var result = (rigidBody.velocity.normalized - collision.GetContact(0).normal) * velocity.magnitude * EnergyMultiplyer;
```
Comment (Kainkun) [inline Assets/Scripts/ArrowReflector.cs:29]: use Vector3.Reflect instead. I don't think you can get a reflection vector without using trig or dot products

**Review by Kainkun — CHANGES_REQUESTED** [review]
Since OnCollisionEnter isnt giving you the correct arrow velocity. Make a prefab variant of the Arrow prefab and give it a script called "ArrowVelocity" and handle velocity that way.
[Heres a link for reference ](https://answers.unity.com/questions/1444252/velocity-data-from-oncollisionenter-is-delayed-inc.html)

File: Assets/Scripts/ArrowReflector.cs (hunk 1 above)
Comment (chmilk2) [reply Assets/Scripts/ArrowReflector.cs:29]: got the reflet function working with a good method that I came up with
Screenshot 2020-10-14 005508
now I'm working on translating it so it follows the reflection angle better

Comment (chmilk2) [reply Assets/Scripts/ArrowReflector.cs:29]: It uses a trigger that gets the velocity of the arrow as it comes near, so the arrow doesn't have to be constantly storing its own velocity

Comment (chmilk2) [reply Assets/Scripts/ArrowReflector.cs]: result
100% complete, arrow reflects and is transformed along the new path

**Review by Kainkun — APPROVED** [review]
wow, this is really advanced. Feels a bit overkill but it's a cool solution.
One thing to consider is that it's under the assumption that arrows will always exit the trigger box but if arrows get destroyed in the trigger or maybe just laying on the ground inside the trigger, it stays in the dictionary which takes up extra memory that won't be used. That shouldn't be an issue for this project though.
Good job!

**Pull request merged**